dashboard/test/controllers/curriculum_proxy_controller_test.rb

```diff
@@ -0,0 +1,26 @@
+require 'webmock/minitest'
+WebMock.disable_net_connect!(allow_localhost: true)
+require_relative '../../../shared/test/spy_newrelic_agent'
+require 'test_helper'
+
+class CurriculumProxyControllerTest < ActionController::TestCase
+  test "should redirect from studio.code.org/docs path to docs.code.org path" do
+    stub_request(:get, "https://docs.code.org/csd/maker_leds/index.html").
+      to_return(body: 'docs.code.org content', headers: {})
+
+    request.host = "studio.code.org"
+    get :get, params: {docs_path: 'csd/maker_leds/index.html'}
+    assert_response :success
+    assert_equal response.body, 'docs.code.org content'
+  end
+
+  test "should redirect from studio.code.org/curriculum path to curriculum.code.org path" do
+    stub_request(:get, "https://curriculum.code.org/csd/maker_leds/index.html").
+      to_return(body: 'curriculum.code.org content', headers: {})
+
+    request.host = "studio.code.org"
+    get :get, params: {curriculum_path: 'csd/maker_leds/index.html'}
```

Comment:
It's still a mystery to me how rails knows that `request.path` here should be `/curriculum/csd/...` instead of `/docs/csd/...`. Not sure if it's because the param I pass is `curriculum_path` instead of `docs_path`. When I had them both just named `path` it seemed to treat both as using the docs_route.

Not sure if there's a better way to write this test.

Reply:
That is mysterious. Something seems fishy here. Does it work for docs too?

This might be better expressed as an [integration test](http://guides.rubyonrails.org/testing.html#integration-testing) where you supply the path. Or if you split up the route into 2 actions as I suggested above then you could test each action independently this way.

In any case we shouldn't need the param anymore, right?

Reply:
The URL is something like `studio.code.org/docs/csd/maker_leds/index.html` so you still need the param to know about the part after docs AFAIK.

Reply:
Oh right. Then this really ought to be an integration test, where we can supply the path and test the routing as well. These [deprecated](https://blog.bigbinary.com/2016/04/19/changes-to-test-controllers-in-rails-5.html) `ActionController::TestCase` functional tests don't test the routing - they just call a controller action directly, more or less, with some context.